### ISSUE TITLE
Feature: (angular) Add Signal support for Angular projects

### ISSUE THREAD

**Issue by bencebkiss** [azat-io/eslint-plugin-perfectionist#355]
### Describe the rule

It would be nice to introduce support for Signals as this is currently not provided. When custom sorting is specified it does not take into account whether the given property has been initialized before being referenced.

### Code example

The sorting by name has been executed.

However since #languages references #selectedLanguageIds the code won't run in this case despite the sorting being correct.

```
  #languages = signal<AddLanguage[]>(
    this.#languagesStore.languages().map((language: Language) => ({
      ...language,
      isSelected: this.#selectedLanguageIds().includes(language.id),
      translatedLang: this.#translocoService.translate(`LANGUAGES.${language.lang}`),
    })),
  );

  #selectedLanguageIds = computed(() => {
    const transformedDocuments = this.#documentsListStore.transformedDocuments();
    const selectedDocument = transformedDocuments.find((document) => document.id === this.#dialogData.documentId);

    return selectedDocument?.languages.map((language) => language.id) ?? [];
  });
```

### Environment

```
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 17.3.11
Node: 20.18.0
Package Manager: npm 10.8.3
OS: win32 x64

Angular: 17.3.12
... animations, common, compiler, compiler-cli, core, elements
... forms, platform-browser, platform-browser-dynamic, router

Package                              Version
--------------------------------------------------------------
@angular-devkit/architect            0.1703.11
@angular-devkit/build-angular        17.3.11
@angular-devkit/core                 17.3.11
@angular-devkit/schematics           17.3.11
@angular/cdk                         17.3.10
@angular/cli                         17.3.11
@angular/material                    17.3.10
@angular/material-date-fns-adapter   17.3.10
@schematics/angular                  17.3.11
rxjs                                 7.8.1
typescript                           5.4.5
zone.js                              0.14.10
```

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @bencebkiss.

The issue comes from the fact that dependencies are currently not detected with `#`members. This will be fixed in v4.0.0.

**Comment by hugop95** [comment]
@bencebkiss You may keep this issue opened until the version is released 😁

**Comment by azat-io** [comment]
Added in 0cf84aef8fd8a33d843e4be445aaa4896e519b67.
Released in [v4.0.0](https://github.com/azat-io/eslint-plugin-perfectionist/releases/tag/v4.0.0).

If you'd like, you can support the release with a retweet:
https://x.com/azat_io_en/status/1858853714238271530
